src/extensions/core/contextMenuFilter.ts

```diff
@@ -8,167 +7,169 @@ const ext = {
-            ?.filter(
-              (w) =>
-                w.type === 'combo' && w.options.values.length === values.length
-            )
-            .find((w) =>
-              w.options.values.every((v, i) => v === values[i])
-            )?.value
-
-          let selectedIndex = clickedComboValue
-            ? values.findIndex((v) => v === clickedComboValue)
-            : 0
-          if (selectedIndex < 0) {
-            selectedIndex = 0
-          }
-          let selectedItem = displayedItems[selectedIndex]
-          updateSelected()
-
-          // Apply highlighting to the selected item
-          function updateSelected() {
-            selectedItem?.style.setProperty('background-color', '')
-            selectedItem?.style.setProperty('color', '')
-            selectedItem = displayedItems[selectedIndex]
-            selectedItem?.style.setProperty(
-              'background-color',
-              '#ccc',
-              'important'
-            )
-            selectedItem?.style.setProperty('color', '#000', 'important')
-          }
-
-          const positionList = () => {
-            const rect = ctx.root.getBoundingClientRect()
-
-            // If the top is off-screen then shift the element with scaling applied
-            if (rect.top < 0) {
-              const scale =
-                1 -
-                ctx.root.getBoundingClientRect().height / ctx.root.clientHeight
-
-              const shift = (ctx.root.clientHeight * scale) / 2
-
-              ctx.root.style.top = -shift + 'px'
-            }
-          }
-
-          // Arrow up/down to select items
-          filter.addEventListener('keydown', (event) => {
-            switch (event.key) {
-              case 'ArrowUp':
-                event.preventDefault()
-                if (selectedIndex === 0) {
-                  selectedIndex = itemCount - 1
-                } else {
-                  selectedIndex--
-                }
-                updateSelected()
-                break
-              case 'ArrowRight':
-                event.preventDefault()
-                selectedIndex = itemCount - 1
-                updateSelected()
-                break
-              case 'ArrowDown':
-                event.preventDefault()
-                if (selectedIndex === itemCount - 1) {
-                  selectedIndex = 0
-                } else {
-                  selectedIndex++
-                }
-                updateSelected()
-                break
-              case 'ArrowLeft':
-                event.preventDefault()
-                selectedIndex = 0
-                updateSelected()
-                break
-              case 'Enter':
-                selectedItem?.click()
-                break
-              case 'Escape':
-                ctx.close()
-                break
-            }
-          })
+    class FilterContextMenu extends ctxMenu {
+      constructor(values: CtxMenuArgs[0], options: CtxMenuArgs[1]) {
```

Comment:
Replace "function + function.prototype" with a extended class that overwrites the constructor with "the function", to get rid of @ts-expect-error.